lib/models/shard.js

```diff
@@ -36,90 +41,41 @@ ShardSchema.set('toObject', {
   transform: function(doc, ret) {
     delete ret.__v;
     delete ret._id;
-
-    doc.contracts = doc.contracts || [];
-    doc.trees = doc.trees || [];
-    doc.challenges = doc.challenges || [];
-    doc.meta =  doc.meta || [];
-
-    ret.contracts = {};
-    ret.trees = {};
-    ret.challenges = {};
-    ret.meta = {};
-
-    for (let i = 0; i < doc.contracts.length; i++) {
-      ret.contracts[doc.contracts[i].nodeID] = doc.contracts[i].contract;
-    }
-
-    for (let i = 0; i < doc.trees.length; i++) {
-      ret.trees[doc.trees[i].nodeID] = doc.trees[i].tree;
-    }
-
-    for (let i = 0; i < doc.challenges.length; i++) {
-      ret.challenges[doc.challenges[i].nodeID] = doc.challenges[i].challenge;
-    }
-
-    for (let i = 0; i < doc.meta.length; i++) {
-      ret.meta[doc.meta[i].nodeID] = doc.meta[i].meta;
-    }
```

Comment:
Ah, gotcha. So you want the data to return as an object like this:

```
{
  hash: 'fjla93fs9-23892-2sdl@#ds-932049203',
  meta: {},
  challenges: {},
  trees: {},
  contracts: { '0': { type: 'contract1' }, '1': { type: 'contract2' } }
}
```

not like this (but it's stored in this format):

```
{
  hash: 'fjla93fs9-23892-2sdl@#ds-932049203',
  meta: [],
  challenges: [],
  trees: [],
  contracts: 
   [  { contract: { type: 'contract1' }, nodeID: '0' },
      { contract: { type: 'contract2' }, nodeID: '1' }
   ]
}
```